### ISSUE TITLE
False positive for NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE with String.substring

### ISSUE THREAD

**Issue by frankgdlc** [spotbugs/spotbugs#2035]
**Problem:** 
Spotbugs finds `NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE` when chaining `java.lang.String.substring` in Java 11 (Corretto), which will never return `null`, per both, the standard [Java SE specs](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html#substring(int,int)) and the [Corretto implementation](https://github.com/corretto/corretto-11/blob/14c02261590b4dc01284888a7a51d39ff581ac8d/src/java.base/share/classes/java/lang/String.java#L1872):

> Returns a string that is a substring of this string. The substring begins at the specified beginIndex and extends to the character at index endIndex - 1. Thus the length of the substring is endIndex-beginIndex.

**Code example:**
```java
String s = fileName.substring(0, 41).replace('_', ':');
```

**Spotbugs report:**
> [ERROR] Medium: Possible null pointer dereference in com.mycompany.myproject.mypackage.MyClass.myMethod() due to return value of called method [com.mycompany.myproject.mypackage.MyClass.myMethod, com.mycompany.myproject.mypackage.MyClass.myMethod] Dereferenced at MyClass.java:[line 19]Known null at MyClass.java:[line 19] NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE
